lalrpop/src/parser/lrgrammar.lalrpop

```diff
@@ -427,6 +430,13 @@ StringLiteral: Atom =
         Ok(Atom::from(text))
     };
 
+StringConstant: String =
```

Comment:
This looks mostly identical to `StringLiteral` above, could you merge the common code into one nonterminal that both calls?

Reply:
I tried this:

```diff
diff --git a/lalrpop/src/parser/lrgrammar.lalrpop b/lalrpop/src/parser/lrgrammar.lalrpop
index 51cb7e0..ab8979b 100644
--- a/lalrpop/src/parser/lrgrammar.lalrpop
+++ b/lalrpop/src/parser/lrgrammar.lalrpop
@@ -424,17 +424,19 @@ QuotedLiteral: TerminalLiteral = {
 };

 StringLiteral: Atom =
-    <lo:@L> <s:"StringLiteral"> =>? {
-        let text = tok::apply_string_escapes(s, lo + 1)
-            .map_err(|e| ParseError::User { error: e })?;
-        Ok(Atom::from(text))
+    <lo:@L> <s:String> =>? {
+        Ok(Atom::from(s))
     };

 StringConstant: String =
+    <lo:@L> <s:String> =>? {
+        Ok(s.into_owned())
+    };
+
+String: Result<Cow<str>, ParseError> =
     <lo:@L> <s:"StringLiteral"> =>? {
-        let text = tok::apply_string_escapes(s, lo + 1)
-            .map_err(|e| ParseError::User { error: e })?;
-        Ok(text.into_owned())
+        tok::apply_string_escapes(s, lo + 1)
+            .map_err(|e| ParseError::User { error: e })?
     };

 RegexLiteral: Atom =
```

but one of the tests started looping. Not sure if this is a lalrpop bug, or what.

Reply:
Interesting.. after `git clean -xfd` it no longer loops, but the test `verify_lalrpop_generates_itself` fails. I don't know what the issue is..

Reply:
Apparently the diff above is not right, the generated .rs does not compile.

In any case, the duplicated code is literally one statement (two lines). I think it's not too bad.